src/nodes/interceptor/eth_send_transaction.go

```diff
@@ -2,19 +2,29 @@ package interceptor
 
 import (
 	"context"
+	"math/big"
 
 	"github.com/consensys/quorum-key-manager/pkg/errors"
-
 	"github.com/consensys/quorum-key-manager/pkg/ethereum"
 	"github.com/consensys/quorum-key-manager/pkg/jsonrpc"
 	proxynode "github.com/consensys/quorum-key-manager/src/nodes/node/proxy"
 	ethcommon "github.com/ethereum/go-ethereum/common"
 )
 
 func (i *Interceptor) ethSendTransaction(ctx context.Context, msg *ethereum.SendTxMsg) (*ethcommon.Hash, error) {
-	i.logger.Debug("sending ETH transaction")
+	switch {
+	case msg.IsPrivate():
+		return i.sendPrivateTx(ctx, msg)
+	case msg.GasPrice != nil:
```

Comment:
I think, even if complexity is low, having a helper function as "isLegacy" might be useful to make our code more verbose and better encapsulate

Reply:
that is not true. The behavior of the standard is: "If the gasPrice is specified, the transaction is a legacy transaction". Whether you specify other params or not is irrelevant.

Reply:
discussed and comment added to explain that pre-london and post-london check